### DISCUSSION TITLE
Can't get multiple sensor data (I'm new in this thing)

### DISCUSSION THREAD

**Discussion by sajjad-hm** [UmerCodez/SensorServer#26] [Q&A] [answered]
**I have tried this code** [Using Multiple Sensors On Single Websocket Connection](https://github.com/umer0586/SensorServer/wiki/Using-Multiple-Sensors-On-Single-Websocket-Connection)

This is what I'm getting output:
connected
connection close
close code :  4002
reason :  Unsupported request

**Comment by UmerCodez** [comment]
@sajjad-hm Make sure that URL is correct i-e ` ws://<ip>:<port>/sensors/connect?types=["<type1>","<type2>","<type3>"...]` note that in this URL its `sensors` and `types` whereas for single sensor you use ` ws://<ip>:<port>/sensor/connect?type=<type>`

**Reply by sajjad-hm** [reply]
Single sensor is working but unfortunately for **multiple sensor**, it's showing this error:

**Output:** 
_connected
connection close
close code :  4002
reason :  Unsupported request_

**Comment by UmerCodez** [comment]
@sajjad-hm Show me your code

**Reply by sajjad-hm** [reply]
```
import websocket
import json


def on_message(ws, message):
    values = json.loads(message)['values']
    type = json.loads(message)['type']
    print("type = ", type)
    print("values = ", values)


def on_error(ws, error):
    print("error occurred")
    print(error)


def on_close(ws, close_code, reason):
    print("connection close")
    print("close code : ", close_code)
    print("reason : ", reason)


def on_open(ws):
    print("connected")


def connect(url):
    ws = websocket.WebSocketApp(url,
                                on_open=on_open,
                                on_message=on_message,
                                on_error=on_error,
                                on_close=on_close)

    ws.run_forever()


connect('ws://192.168.0.100:8081/sensors/connect?types=["android.sensor.accelerometer","android.sensor.gyroscope"]')

```

**Reply by UmerCodez** [reply]
Your URL seems correct
 `connect('ws://192.168.0.100:8081/sensors/connect?types=["android.sensor.accelerometer","android.sensor.gyroscope"]')` 
Still I am unable to find whats wrong !!

**Comment by UmerCodez** [comment]
Use `ws://192.168.0.100:8081/sensors/connect?types=["android.sensor.accelerometer","android.sensor.gyroscope"]` in http://livepersoninc.github.io/ws-test-page/ and see if it works

**Reply by sajjad-hm** [reply]
Still have the Same problem, It can't connect when I try multiple sensor data but on this test page also single sensor data stream is working perfectly. 

I can't figure out where the problem is happening.

**Reply by UmerCodez** [reply, marked as answer]
That is weird ! It works on my phone I tried `connect('ws://192.168.0.102:8080/sensors/connect?types=["android.sensor.accelerometer","android.sensor.gyroscope"]')`

**Reply by sajjad-hm** [reply]
Thanks for the help bro. I have solved the problem. Unfortunately, I was using an old version of this app. Installing the latest version, it solved the problem. I apologize for the inconvenience.

**Reply by UmerCodez** [reply]
If you are trying to read only `accelerometer` and `gyroscope` you can write multi threaded python application where on thread receives accelerometer data and other thread receives gyroscope data

**Reply by sajjad-hm** [reply]
I'm at the beginner level right now. I'll try that. Still, I have to figure out lots of new things.
